src/colorizer/Dataset.ts

```diff
@@ -246,21 +348,36 @@ export default class Dataset {
     const manifest = await manifestLoader(this.manifestUrl);
 
     this.frameFiles = manifest.frames;
-    this.featureFiles = manifest.features;
     this.outlierFile = manifest.outliers;
     this.metadata = { ...defaultMetadata, ...manifest.metadata };
 
-    const featuresToMetadata: Record<string, Partial<FeatureMetadata>> = {};
-    for (const featureName of Object.keys(this.featureFiles)) {
-      featuresToMetadata[featureName] = manifest.featureMetadata ? manifest.featureMetadata[featureName] : {};
-    }
-
     this.tracksFile = manifest.tracks;
     this.timesFile = manifest.times;
     this.centroidsFile = manifest.centroids;
 
     this.frames = new FrameCache(this.frameFiles.length, MAX_CACHED_FRAMES);
-    const featuresPromises: Promise<void>[] = this.featureNames.map((name) =>
+
+    // Load feature data -> switch between deprecated and new feature type loading.
+    let featuresToMetadata: Record<string, FeatureMetadata> = {};
+    if (this.isFeatureDeprecated(manifest.features)) {
```

Comment:
based on my earlier comment, this could look something like `manifest.features = updateManifestVersion(manifest.features, manifest.featureMetadata)` .. roughly speaking